_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+
+![processes](/img/ssdlc/processes.jpg)
+
+So what happened to build the software and let security engineers test it afterwards? The same happened as with all the other testing too. It is much cheaper to catch mistakes early. Security problems in requirements and/or in design are pretty expensive to fix if you let them go to the production. You want to fix them early. To catch them early you need supporting routines to have a systematic approach in finding them. That is what SSDLC is all about. 
+
+## How to implement secure software development lifecycle
+
+Start small and make it evolve over time. Treat the improvements in SSDLC like you would treat any other process improvements or quality assurance tasks. Make them as a visible for in your agile boards and make a supporting document that clearly tells what you are after for.
+
+#### Requirements
+
+There are laws and regulations. There are company policies and customer policies. Failing to notice these will make it difficult to deal with them later. Gathering intelligence on what you are about to do businesswise should make you able to also make some security requirements for the project. This will create the foundation of your security awareness level later on. Are you dealing with credit cards? Surely you have then heard about PCI-DSS. Are you creating an application for healthcare? Maybe you ought to know something about HIPAA. Know your industry and make use of whatever given requirements there are for the field. You should also have a hunch about data classification levels.
+
+To sum up: 
+* Know the laws and regulations
+* Follow company policies for your own company and for your customers company
+* Check for useful standards
+* Make security requirements
+* Data classification
+
+#### Design
+
+Design sets a foundation for software architecture. Security was all about CIA (confidentiality, integrity and availability). Failures in software architecture most often affects one of the letters of CIA and they are also often pretty expensive to fix. To fight this we would like to setup extra care for design. There are plenty of tools and and practices in different software project management frameworks but I would like to bring out threat modeling. There are different approaches for threat modeling but what I have found useful is to create diagrams for software architecture from different approaches: deployment pipeline, infrastructure and networking, sequence diagrams and workflows. Reviewing those diagrams from the "how would I hack this?" -perspective is really useful. You can also go so far that you will create evil user stories for the backlog. 
```

Comment:
"Security was all about CIA" mulle ei auennut miksi tässä on mennyt aikamuoto. Onko tarkoituksella vai ei?

Reply:
ehkä mä vaihdan sen vaan is, ei se nyt oo mitään historiallista tietoa. Vähän vaan luennoitseva tapa kertoa miten maailmanlaita on.